stateoftheuniverse/widgets/moon_functions.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/python3
+"""
+A function to return moon phase & moon brightness.
+"""
+
+import numpy as np
+import astroplan
+import astropy
+from astropy.time import Time
+import matplotlib.pyplot as plt
+from astropy.coordinates import get_moon, get_sun
+import datetime
+import warnings
+warnings.filterwarnings('ignore')
+
+def get_moon_illumination(time: datetime) -> float:
+    """ A function to return the moon illumination
+
+    I'll fill this in later
+    """
+
+    astropytime = Time(time)
+    try:
+        illumination = astroplan.moon.moon_illumination(astropytime)
+    except:
+        print("Something has gone wrong. Please raise an issue on GitHub")
+    return illumination*100
+
+def get_moon_state(time: datetime) -> bool:
+    """ A function to return whether the moon is waxing or waning
+    """
+    astropytime = Time(time)
+    moonra = get_moon(time).ra.value
+    sunra = get_sun(time).ra.value
+    opposite = sunra + 180.
+    if opposite > 360.:
+        opposite -= 360.
+
+    #Moon goes for west to east
+    #Find whether waxing or waning
+    if sunra > 180:
+        if (moonra > sunra) or (moonra < opposite):
+            waxing = True
+        else:
+            waxing=False
+
+    elif sunra <= 180:
+        if (moonra > sunra) & (moonra < opposite):
+            waxing = True
+        else:
+            waxing=False
+
+def get_moon_phase(time: datetime) -> str:
+    """ Code to get moon phase"""
+    illumination = get_moon_illumination(time)
+    waxing = get_moon_state(time)
+
+    if  illumination <= 0.1:
```

Comment:
This following section could maybe be rewritten a little more concisely as:
```python
if illumination <= 0.1:
    phase = 'New Moon'
elif 0.1 < illumination < 49.9:
    phase = 'Waxing Crescent' if waxing else 'Waning Crescent'
elif 49.9 <= illumination <= 50.1:
    phase = 'First Quarter' if waxing else 'Last Quarter'
elif 50.1 < illumination < 99.9:
    phase = 'Waxing Gibbous' if waxing else 'Waning Gibbous'
else
    phase = 'Full Moon'
```
Also, are you sure about when it needs to be `<` and when it should be `<=`? The current version seems a little inconsistent, but perhaps that's just how it is actually defined?

Reply:
Full moon, the quarters, and new moon are at 100%, 50% and 0% illumination respectively, but to avoid any issues due to small floating errors I wanted to give those phase classifications the extra 0.1% leeway.

Thanks for the great suggestion! That's super neat Python code, I love learning new ways to make my code more compact! :)